### PR TITLE
docs(adr): ADR-20: Post ownership transfer

### PR DESCRIPTION
**Pull request by dadamu**
## Description

This PR represents the ADR for a new feature that will allow users to transfer the ownership of their posts to another user.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://docs.cosmos.network/v0.44/building-modules/intro.html)
- [ ] included the necessary unit and integration [tests](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: docs/architecture/adr-020-post-ownership-change.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  string owner = 14;
+}
+```
+
+### `Msg` Service
+
+```proto
+service Msg {
+    // ChangePostOwner allows users to transfer the ownership of a post to another person
+    rpc ChangePostOwner(MsgChangePostOwner) returns (MsgChangePostOwnerResponse);
```
Comment (RiccardoM) [inline docs/architecture/adr-020-post-ownership-change.md:43]: Can't we simply use the `MsgEditPost` message instead? Do we have any advantage in defining a new message type over using the existing one?

Comment (dadamu) [reply docs/architecture/adr-020-post-ownership-change.md:43]: Having a new message can follow SRP(Single responsibility principle), make sure `MsgEditPost` only modify the post content, which brings the benefit as follows:
1. User can avoid move ownership accidentally when editing a post, and vice versa
2. `MsgEditPost` related methods inside `desmos-bindings`, `desmjs` can stay away from breaking change
3. Tracker can easily track the ownership changes history of a post

As result, I decided to define a new message.

**Review by RiccardoM — APPROVED** [review]
Looks good to me. Please change its status to be `Accepted Not Implemented`

**Comment by dadamu** [comment]
@RiccardoM Updated

**Pull request merged**